pipeline/04-interpret.R

```diff
@@ -31,10 +31,32 @@ if (shap_enable || comp_enable) {
   # PINs) that need values. Will use the the trained model to calc SHAP values
   assessment_data <- as_tibble(read_parquet(paths$input$assessment$local))
 
+  # Aggregate square footage to building level. We do this to ensure consistent
+  # shap values and comps based on the generated prediction for multi-card pins.
+  # More details in multi-card handling in the assess stage.
+  # https://github.com/ccao-data/model-res-avm/issues/358
+
+  # Persist sort order
+  assessment_data_ordered <- assessment_data %>%
+    group_by(meta_pin) %>%
+    arrange(desc(char_bldg_sf), meta_card_num) %>%
+    mutate(sqft_card_num_sort = row_number()) %>%
```

Comment:
This column `sqft_card_num_sort` is persisted throughout the shaps and comps calculations to maintain the correct ordering